tilavarauspalvelu/integrations/email/find_recipients.py

```diff
@@ -117,3 +117,40 @@ def get_reservation_staff_notification_recipients_by_language(
             recipients_by_language[user_language].add(user.email.lower())
 
     return recipients_by_language
+
+
+def get_application_section_staff_notification_recipients_by_language(
+    application_section: ApplicationSection,
+) -> dict[Lang, set[str]]:
+    """
+    Get staff users who should receive email notifications for the given application.
+    Group recipients by their preferred language.
+    """
+    recipients_by_language: dict[Lang, set[str]] = defaultdict(set)
+
+    notification_settings = [ReservationNotification.ALL]
+
+    # Only fetch users with active unit roles and appropriate notification settings.
+    users = User.objects.filter(
+        unit_roles__isnull=False,
+        unit_roles__role_active=True,
+        reservation_notification__in=notification_settings,
+    ).exclude(email="")
+
+    # Skip the reservation creator
+    if application_section.application.user:
+        users = users.exclude(pk=application_section.application.user.pk)
+
+    units = (
+        Unit.objects.prefetch_related("unit_groups")
+        .filter(reservation_units__reservation_unit_options__application_section=application_section)
```

Comment:
The reservation units need to be fetched from the actual reservation series generated from the allocations, since original options set while applying may contain ones for which nothing was allocated in application handling.

Reply:
Very good point, fixed this :)